sda-commons-app-example/src/test/java/org/sdase/commons/spring/boot/tracing/KafkaTraceTokenIT.java

```diff
@@ -0,0 +1,155 @@
+
+  private BlockingQueue<ConsumerRecord<String, String>> consumerRecords;
+
+  @Value("${app.kafka.producer.topic}")
+  private String topic;
+
+  @BeforeEach
+  void setUp() {
+
+    WireMock.reset();
+
+    traceTokenRecordInterceptor = new TraceTokenRecordInterceptor();
+
+    Map<String, Object> configs =
+        new HashMap<>(KafkaTestUtils.consumerProps("consumer", "false", embeddedKafkaBroker));
+
+    DefaultKafkaConsumerFactory<String, String> consumerFactory =
+        new DefaultKafkaConsumerFactory<>(
+            configs, new StringDeserializer(), new StringDeserializer());
+
+    ContainerProperties containerProperties = new ContainerProperties(topic);
+    listenerContainer = new KafkaMessageListenerContainer<>(consumerFactory, containerProperties);
+
+    consumerRecords = new LinkedBlockingQueue<>();
+    listenerContainer.setupMessageListener(new KafkaTraceTestMessageListener<>(consumerRecords));
+    listenerContainer.setRecordInterceptor(traceTokenRecordInterceptor);
+    listenerContainer.start();
+    ContainerTestUtils.waitForAssignment(
+        listenerContainer, embeddedKafkaBroker.getPartitionsPerTopic());
+  }
+
+  @Test
+  void shouldPreserveTraceTokenIfPresent() throws InterruptedException {
+    // given
+    var headers = new HttpHeaders();
+    headers.set("Trace-Token", "pre-defined-trace-token");
+    // when
+    ResponseEntity<String> responseEntity =
+        executeRequestWithHeader("/api/createEvent?user=user_1", port, headers);
+
+    ConsumerRecord<String, String> poll = consumerRecords.poll(10, TimeUnit.SECONDS);
+
+    Optional<Header> headerParentTraceToken =
+        Arrays.stream(poll.headers().toArray())
+            .filter(header -> header.key().equals("Parent-Trace-Token"))
+            .findFirst();
+
+    assertTrue(headerParentTraceToken.isPresent());
+    assertThat(new String(headerParentTraceToken.get().value()))
+        .isEqualTo("pre-defined-trace-token");
+
+    String mdcTraceTokenAfterMessageConsumed = MDC.get("Parent-Trace-Token");
+
+    assertThat(mdcTraceTokenAfterMessageConsumed).isNull();
+  }
+
+  private ResponseEntity<String> executeRequestWithHeader(
+      String path, int port, HttpHeaders headers) {
+    return client.exchange(
+        String.format("http://localhost:%d%s", port, path),
+        HttpMethod.GET,
+        new HttpEntity<>(headers),
+        String.class);
+  }
+
+  class KafkaTraceTestMessageListener<K, V> implements MessageListener<K, V> {
+
+    private BlockingQueue<ConsumerRecord<K, V>> consumerRecords;
+
+    public KafkaTraceTestMessageListener(BlockingQueue<ConsumerRecord<K, V>> consumerRecords) {
+      this.consumerRecords = consumerRecords;
+    }
+
+    @Override
+    public void onMessage(ConsumerRecord<K, V> data) {
+      String mdcTraceTokenWhileMessageConsumed = MDC.get("Parent-Trace-Token");
+      assertThat(mdcTraceTokenWhileMessageConsumed).isEqualTo("pre-defined-trace-token");
```

Comment:
NIT: Having the assertion here feels quite unrelated to the test. An alternative would be add the value found in the MDC with all other information that must be asserted in the result list and move the assertion to the test. I'm always a bit afraid, that this line of code might not be hit. Also I assume, it will not cause a test failure as the message handling might happen in another thread. Afaik, that is not the case with the direct call of `poll` in line 115, but bith scenarios may be easily missed in a refactoring.

Reply:
moved assertion to test